src/services/carts.service.ts

```diff
@@ -0,0 +1,95 @@
+import { carts } from "../dataAccess/carts";
+import { CartEntity, CartItemEntity } from "../models/cart.entity";
+import { v4 as uuidv4 } from "uuid";
+import { ProductEntity } from "../models/product.entity";
+import { findProduct } from "./products.service";
+import { Response } from "express";
+import Joi from "joi";
+
+export const createCart = (userId: string) => {
+  const userCart = {
+    id: uuidv4(),
+    userId: userId,
+    isDeleted: false,
+    items: [],
+  };
+  carts.push(userCart);
```

Comment:
Service layer should not know implementation details of data layer.
It should be like `repository.createCart(userCart)`